### PR TITLE
feat: surface per-object recovery commands in dashboard

### PR DESCRIPTION
**Pull request by JuaniRios**
## What
Implements [RAI-987](https://linear.app/makeitrain/issue/RAI-987) — Step 9 of [RAI-978](https://linear.app/makeitrain/issue/RAI-978). Turns the dashboard detail modals into a recovery command palette and adds a CLI recovery guide.

## Why
The dashboard only surfaced a recovery command in one narrow case (the "Stranded equity" hint, and only once a transfer reached a recognized failed/stranded terminal). A transfer stuck mid-flight but not yet failed showed no actionable command, so operators had to recall the exact `stox` invocation from memory or hunt through docs.

## How
- `dashboard/src/lib/transfer.ts`: replaces the single `recoveryCommand` with per-object generators (`transferRecoveryCommands`, `tradeRecoveryCommands`) returning every command valid for the object's `kind`+`status` — including the stuck-but-not-failed case — each as `{ command, label, description, mode }` with the **new** grouped command names (`transfer recheck|resume|fail|reconcile --kind ...`, `position release-hedge|set`, `view rebuild`, `process-tx`). Keeps the deployment prefix logic (`stox` vs sim `nix develop … cli --`).
- Transfer + trade detail modals render a "CLI commands" section: one copy-paste block per applicable command with its execution-mode badge (direct-DB vs requires-bot).
- New `cli-command-block.svelte` (reusable copy block) and `recovery-guide.svelte` (persistent button → dialog documenting every command grouped by object), wired into `header-bar.svelte`. The modals and guide read from the same `transfer.ts` source.

## Testing
17 new/updated assertions in `transfer.test.ts` (per-kind/per-status command sets, exact new-name strings, prod+sim prefix behavior, exec-mode flags). `bun run test:run` → 143 passed; eslint/prettier clean on all changed files.

## Anything else
The 4 `svelte-check` errors on this branch are pre-existing `alpacaUsdc` generated-DTO drift in `inventory-cash.*`/`connection.svelte.test.ts` (present on master, files untouched here; CI's DTO-regen step resolves them locally-stale types) — unrelated to this change. `position release-hedge` shows an `<order-id>` placeholder since the trade modal has no offchain order id. Completes epic [RAI-978](https://linear.app/makeitrain/issue/RAI-978).

### CODE REVIEW

**Comment by JuaniRios** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/815?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#815** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/815?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 2  dependent PRs ([#818](https://github.com/ST0x-Technology/st0x.liquidity/pull/818) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/818?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#935](https://github.com/ST0x-Technology/st0x.liquidity/pull/935) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/935?utm_source=stack-comment-icon" target="_blank">Graphite</a>) 👈 <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/815?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#813** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/813?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#812** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/812?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#811** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/811?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#810** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/810?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#809** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/809?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#808** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/808?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#807** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/807?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#806** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/806?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#781** <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/781?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#782](https://github.com/ST0x-Technology/st0x.liquidity/pull/782) <a href="https://app.graphite.com/github/pr/ST0x-Technology/st0x.liquidity/782?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>add-to-gt-merge-queue</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/ST0x-Technology?ref=merge-queue-instructions-in-stack-comment&prId=815).

<h2></h2>
<sub>

An organization admin has required the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Comment by JuaniRios** [comment]
@coderabbitai full review

**Comment by JuaniRios** [comment]
@coderabbitai full review

File: src/usdc_rebalance.rs [hunk 1]
```diff
@@ -714,6 +714,28 @@ pub(crate) enum UsdcRebalance {
...
+            Self::DepositFailed { .. }
+            | Self::ConversionFailed {
+                direction: RebalanceDirection::BaseToAlpaca,
+                ..
+            } => true,
+            Self::BridgingFailed {
+                burn_tx_hash,
+                cctp_nonce,
+                ..
+            } => burn_tx_hash.is_some() || cctp_nonce.is_some(),
```
Comment (0xgleb) [inline src/usdc_rebalance.rs:734]: should fix: this `BridgingFailed` branch is the most regression-prone path in `is_post_burn_failure`, a two-field OR over `burn_tx_hash`/`cctp_nonce`, yet no `to_dto` test exercises it. The PR only covers `DepositFailed` (true) and `WithdrawalFailed` (false via the catch-all), so a future field rename or wrong field tested would silently flip `post_burn` and break the dashboard reconcile gating with no failing test. Add `to_dto` tests for `BridgingFailed { burn_tx_hash: Some, cctp_nonce: None }` (post_burn true) and both-None (post_burn false), plus `ConversionFailed { BaseToAlpaca }` (true) and `AlpacaToBase` (false).

File: src/usdc_rebalance.rs [hunk 2]
```diff
@@ -714,6 +714,28 @@ pub(crate) enum UsdcRebalance {
...
+            _ => false,
```
Comment (0xgleb) [inline src/usdc_rebalance.rs]: should fix: this `_ => false` catch-all defeats the compiler's variant check on the owned `UsdcRebalance` enum, which is exactly what the doc comment's "MUST stay in sync with `transition_reconcile_stuck_rebalance`" relies on. A future post-burn failure variant would silently fall through to `false` and diverge from the reconcilable set with no compile error, and the repo guideline requires exhaustive matches in production so new variants are flagged. Replace the wildcard with explicit `false` arms for the remaining in-flight and non-reconcilable failure variants so adding a variant forces a conscious classification.

File: src/usdc_rebalance.rs [hunk 3]
```diff
@@ -714,6 +714,28 @@ pub(crate) enum UsdcRebalance {
 }
 
 impl UsdcRebalance {
+    /// Whether a failed rebalance failed POST-burn -- the only USDC failures
+    /// `transfer reconcile --kind usdc` accepts (a pre-burn failure strands
+    /// nothing and the CLI rejects it). MUST stay in sync with the reconcilable
+    /// set in `transition_reconcile_stuck_rebalance`. A new (non-failure or
+    /// future) state defaults to `false`, i.e. not reconcilable -- the safe
+    /// default for surfacing the command on the dashboard.
+    fn is_post_burn_failure(&self) -> bool {
```
Comment (0xgleb) [inline src/usdc_rebalance.rs:723]: minor: `is_post_burn_failure` must stay aligned with both `transition_reconcile_stuck_rebalance` and the CLI reconcile preflight, but the only `to_dto` tests cover `DepositFailed` and `WithdrawalFailed`. The discriminating cases, pre-burn vs post-burn `BridgingFailed` and `ConversionFailed` by direction, go through `holds_rebalance_guard` tests on a separate match, so a refactor of one arm without the other would not be caught here. A small set of parametrized `to_dto` tests over each reconcilable/non-reconcilable failure variant would close the gap.

File: crates/dto/src/transfer.rs [hunk 4]
```diff
@@ -330,7 +344,16 @@ mod tests {
         assert!(
             bridge_operation(UsdcBridgeStatus::Completed { completed_at: now }, now).is_terminal()
         );
-        assert!(bridge_operation(UsdcBridgeStatus::Failed { failed_at: now }, now).is_terminal());
+        assert!(
+            bridge_operation(
+                UsdcBridgeStatus::Failed {
+                    failed_at: now,
+                    post_burn: true
```
Comment (0xgleb) [inline crates/dto/src/transfer.rs:388]: minor: this test only passes `post_burn: true` to the `Failed` variant, so it never confirms that a pre-burn failure (`post_burn: false`) is also terminal. `is_terminal()` should be true regardless of `post_burn` because that field gates a separate decision (reconcile availability), and if someone narrowed the pattern to `Failed { post_burn: true, .. }` this test would still pass. Add a matching assert with `post_burn: false` right after this one.

File: crates/dto/src/transfer.rs [hunk 5]
```diff
@@ -122,8 +122,22 @@ pub enum UsdcBridgeStatus {
...
+        // True when the failure happened post-burn (`DepositFailed`, a post-burn
+        // `BridgingFailed`, or a `BaseToAlpaca ConversionFailed`) -- the only USDC
+        // failures `transfer reconcile --kind usdc` accepts. False for pre-burn
+        // failures, which strand nothing and the CLI rejects, so the dashboard can
+        // offer reconcile per-object only when this is true. NOTE: a plain `//`
+        // comment, not `///` -- ts-rs emits doc comments into the generated TS
+        // binding, and a field-level doc lands awkwardly inside the object type
+        // literal (`{ failedAt: string, /** ... */ postBurn: boolean }`), which
+        // trips the dashboard's type-aware eslint. Keep the rationale here in Rust.
+        post_burn: bool,
```
Comment (0xgleb) [inline crates/dto/src/transfer.rs:139]: minor: adding `post_burn` to `Failed` introduces no serialization roundtrip test asserting the API shape (`{ status: "failed", failedAt, postBurn }`) that the dashboard reads via `transfer.status.postBurn`. The reconcile gating depends on `postBurn` surviving JSON plus ts-rs codegen with the camelCase rename, and a serde tagging mistake would silently show or hide reconcile for all failed bridges. Add a test asserting exact `serde_json::json!` literals for `post_burn: true` and `false`.

Comment (0xgleb) [inline crates/dto/src/transfer.rs:139]: minor: `post_burn: bool` encodes a domain distinction (pre-burn vs post-burn failure, which gates whether `transfer reconcile --kind usdc` is offered) as a raw boolean, which the repo's "avoid boolean blindness" guideline discourages for domain states. A two-variant enum like `BurnPhase { PreBurn, PostBurn }` would make the meaning legible at every construction and match site and serialize via ts-rs as a string-literal union instead of a bare bool. This is a DTO-boundary judgement call since the bool maps cleanly to the existing `postBurn?: boolean` the dashboard consumes, so it is acceptable as-is, but worth considering.

File: dashboard/src/lib/components/trade-history-panel.svelte [hunk 6]
```diff
@@ -676,6 +682,29 @@
             </div>
           {/each}
         </div>
+
+        {@const txHash =
+          trade.venue === 'raindex' && trade.id.includes(':')
+            ? trade.id.slice(0, trade.id.lastIndexOf(':'))
+            : null}
+        {@const recoveryCommands = tradeRecoveryCommands({
```
Comment (0xgleb) [inline dashboard/src/lib/components/trade-history-panel.svelte:686]: should fix: `tradeRecoveryCommands` is rendered for every trade detail modal with no gating on trade state, unlike `transferRecoveryCommands` which gates on status. For raindex fills this pre-fills `process-tx --tx-hash <hash>` (plus `position set`/`release-hedge`/`view rebuild`) on trades that were already successfully processed. Re-running `process-tx` on an older hash only gets rejected when the trade is still `last_acknowledged_trade_id`, so on a routine history view it is one click away from re-applying a fill and inflating net exposure. Gate these on actionable state (missed fill / stuck hedge), or at minimum drop per-trade `process-tx` unless the trade is known unaccounted.

File: dashboard/src/lib/transfer.ts [hunk 7]
```diff
@@ -137,55 +147,446 @@ export const stuckReasonLabel = (reason: string): string =>
...
+  const prefix = commandPrefix(params.deployment)
+  if (prefix === null) return []
+
+  const { symbol, txHash } = params
+
+  const commands: RecoveryCommand[] = []
+
+  if (txHash !== null) {
+    commands.push({
+      command: `${prefix} process-tx --tx-hash ${txHash}`,
```
Comment (0xgleb) [inline dashboard/src/lib/transfer.ts]: should fix: this surfaces `process-tx` as a prominent copy-paste button, but the underlying `process_found_trade` path goes straight to `AcknowledgeOnChainFill` without consulting the `OnChainTrade` aggregate or the full set of acknowledged trade ids. Duplicate detection is only `last_acknowledged_trade_id == trade_id`, a single-slot guard, so a retry after a later fill was acknowledged double-applies the older trade. Before emitting `AcknowledgeOnChainFill`, check the persisted `OnChainTrade` / fill witness for `(tx_hash, log_index)` and fail closed if already recorded, and align the UI copy with the actual idempotency guarantee.

File: dashboard/src/lib/transfer.ts [hunk 8]
```diff
@@ -137,55 +147,446 @@ export const stuckReasonLabel = (reason: string): string =>
...
-      command: `nix develop --command cargo run --features mock --bin cli -- --config ${basePath}.config.toml --secrets ${basePath}.secrets.toml transfer recheck --kind ${transferType} --id ${params.id}`,
+  if (isTerminalStatus(status)) return []
+
+  // The CLI requires --direction and rejects a mismatch against the persisted
+  // value, so fill the bridge's known direction (translated into the CLI's flag
+  // vocabulary) when we have it rather than leaving a placeholder.
+  const directionArg = usdcDirectionToCliFlag(direction) ?? '<to-raindex|to-alpaca>'
+  return [
+    {
+      command: `${prefix} transfer resume --kind usdc --id ${id} --direction ${directionArg}`,
```
Comment (0xgleb) [inline dashboard/src/lib/transfer.ts:422]: should fix: this resume command is offered for any non-terminal, non-failed status, which includes pre-burn stages `converting` and `withdrawing` where the bot is still actively driving the Alpaca withdrawal. The description says the bridge was "interrupted after the burn," so it is misleading for those statuses, and the `direct-db-live-rpc` mode means it hits live chain and can race the running bot. Narrow resume to post-burn statuses (`bridging`, `depositing`) and consider surfacing `fail-usdc-transfer` for pre-burn stuck bridges, which currently has no per-object affordance.

File: dashboard/src/lib/transfer.ts [hunk 9]
```diff
@@ -137,55 +147,446 @@ export const stuckReasonLabel = (reason: string): string =>
...
+      },
+      {
+        command: 'stox transfer fail --kind <mint|redemption> --id <id> -r "<reason>"',
+        description: 'Force a stuck transfer into the terminal Failed state.',
+        whenToUse: 'A mint/redemption is permanently stuck and unrecoverable.',
+        appliesTo: 'Equity mint / redemption (non-terminal)',
+        mode: 'direct-db'
+      },
+      {
+        command: 'stox transfer reconcile --kind <usdc|mint|redemption> --id <id> -r "<reason>"',
```
Comment (0xgleb) [inline dashboard/src/lib/transfer.ts:551]: minor: the guide documents equity `fail` and USDC `resume`/`reconcile` but omits `fail-usdc-transfer`, the documented operator path for clearing pre-burn USDC stuck states (`BridgingSubmitting`, `WithdrawalComplete`). Operators relying on this guide get no guidance for that known stuck state, and the per-object generator omits it too. Add a guide entry (and optionally per-object gating) for the pre-burn USDC fail command with the same execution-mode labeling.

File: crates/dto/src/transfer.rs (hunk 4 above)
Comment (0xgleb) [inline crates/dto/src/transfer.rs:388]: minor: the Rust `Failed` variant now carries `post_burn`, but the generated `dashboard/src/lib/api/UsdcBridgeStatus.ts` (not in this diff) still shows `{ status: "failed", failedAt }` without `postBurn`, so it will fail svelte-check until CI regenerates DTOs. Nothing currently imports the stale generated type (`transfer-panel.svelte` uses a local ad-hoc shape), so this is developer confusion rather than a runtime break, but the drift count grows. Run the DTO regeneration step locally and commit the regenerated `UsdcBridgeStatus.ts` so the type is self-documenting.

File: dashboard/src/lib/components/recovery-guide.svelte [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+                <div class="mt-1.5 grid grid-cols-[auto_1fr] gap-x-3 gap-y-1 text-[11px]">
+                  <span class="text-muted-foreground">What</span>
+                  <span>{entry.description}</span>
+                  <span class="text-muted-foreground">When</span>
+                  <span>{entry.whenToUse}</span>
+                  <span class="text-muted-foreground">Applies to</span>
+                  <span>{entry.appliesTo}</span>
+                  <span class="text-muted-foreground">Mode</span>
+                  <span
+                    class={entry.mode === 'requires-bot' ? 'text-amber-500' : 'text-destructive'}
```
Comment (0xgleb) [inline dashboard/src/lib/components/recovery-guide.svelte]: minor: this badge uses only two colors, amber for `requires-bot` and red for everything else, so `live-rpc-only`, `direct-db-live-rpc`, and `direct-db` all render identically red. `live-rpc-only` (`cctp complete-mint`) writes no DB state and is safe to run while the bot is up, yet it gets the same warning color as `view rebuild` which mandates stopping the bot. The text label via `recoveryModeLabel` does distinguish them, but consider a third color for `live-rpc-only` (and the same fix in `cli-command-block.svelte`) so the badge does not imply equal severity.

File: dashboard/src/lib/components/recovery-guide.svelte [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+      onclick={() => dialogEl?.close()}
+    >
+      &times;
+    </button>
+  </div>
+
+  <div class="max-h-[70vh] overflow-y-auto px-5 py-4">
+    <p class="mb-4 text-xs text-muted-foreground">
+      Every recovery command, grouped by object. The <span class="font-mono">stox</span> prefix is the
+      production wrapper (run on the server via Tailscale ssh). Each command notes whether it mutates
```
Comment (0xgleb) [inline dashboard/src/lib/components/recovery-guide.svelte:54]: minor: `RECOVERY_GUIDE` hardcodes the `stox` prefix and this component renders it with no deployment context, so in a simulation build (`PUBLIC_SIMULATE_SOURCE_ID` set) the guide shows `stox ...` commands that a sim operator cannot run; they need the mock-cli prefix the per-object modals already apply. This prose note partly mitigates it, but a sim operator may still copy a guide command and hit a CLI error. Either make the production-only caveat visually prominent when `isDashboardMockMode()` is true, or pass deployment context in and render via `commandPrefix`.

File: dashboard/src/lib/components/transfer-panel.svelte [hunk 12]
```diff
@@ -48,13 +51,13 @@
...
-    kind: string
+    kind: TransferCategory
     id: string
     symbol?: string
     quantity?: string
     amount?: string
-    direction?: string
-    status: { status: string }
+    direction?: UsdcBridgeDirection
+    status: { status: string; postBurn?: boolean }
```
Comment (0xgleb) [inline dashboard/src/lib/components/transfer-panel.svelte:61]: minor: widening `TransferEntry.status` to `{ status: string; postBurn?: boolean }` exposes `postBurn` on every transfer kind at the type level, even though it is only meaningful for `kind === 'usdc_bridge'`. Routing is correct at runtime because `transferRecoveryCommands` dispatches on `kind` before consulting `postBurn`, so this is a maintainability concern, not a live bug. Consider narrowing the status to a union that carries `postBurn` only for USDC bridges

**Review by 0xgleb — CHANGES_REQUESTED** [review]

File: src/usdc_rebalance.rs (hunk 1 above)
Comment (JuaniRios) [reply src/usdc_rebalance.rs:734]: Added four `to_dto` tests covering both `BridgingFailed` OR-branches (burn-tx present / absent) and both `ConversionFailed` directions. 35f404ca

File: src/usdc_rebalance.rs (hunk 2 above)
Comment (JuaniRios) [reply src/usdc_rebalance.rs]: Replaced the `_ => false` wildcard with exhaustive arms over all 17 variants, so a new variant forces a conscious post-burn classification instead of silently defaulting to false. 35f404ca

File: src/usdc_rebalance.rs (hunk 3 above)
Comment (JuaniRios) [reply src/usdc_rebalance.rs:723]: Covered by the four new `to_dto_{bridging,conversion}_failed_*` tests, which exercise both OR-branches and both directions through `to_dto` — matching the file's individual-named-test convention. 35f404ca

File: crates/dto/src/transfer.rs (hunk 4 above)
Comment (JuaniRios) [reply crates/dto/src/transfer.rs:388]: Added the `Failed { post_burn: false }` terminality assertion (post_burn gates reconcile, not terminality). 35f404ca

File: crates/dto/src/transfer.rs (hunk 5 above)
Comment (JuaniRios) [reply crates/dto/src/transfer.rs:139]: Added `usdc_bridge_failed_serializes_post_burn_flag` asserting the `failed` / `postBurn` JSON shape with `json!()` literals (both true and false). 35f404ca

Comment (JuaniRios) [reply crates/dto/src/transfer.rs:139]: Leaving as `bool`: it maps 1:1 to the dashboard's existing `postBurn?: boolean` consumer at a pure serialization boundary, and the field comment documents the pre/post-burn meaning. A `BurnPhase` enum would change the wire shape and the consumer for no behavioural benefit.

File: dashboard/src/lib/components/trade-history-panel.svelte (hunk 6 above)
Comment (JuaniRios) [reply dashboard/src/lib/components/trade-history-panel.svelte:686]: Removed `process-tx` from the per-trade modal — a trade only reaches the history panel once recorded, so the modal could only ever re-apply an accounted fill, and the Position single-slot `last_acknowledged_trade_id` guard doesn't catch a re-run on an older trade. It stays in the static recovery guide as CLI reference. 35f404ca

File: dashboard/src/lib/transfer.ts (hunk 8 above)
Comment (JuaniRios) [reply dashboard/src/lib/transfer.ts:422]: Narrowed `resume` to post-burn (`bridging`/`depositing`); `converting`/`withdrawing` are pre-burn and now offer only `fail-usdc-transfer` (strands nothing, doesn't race the bot). 35f404ca

File: dashboard/src/lib/transfer.ts (hunk 9 above)
Comment (JuaniRios) [reply dashboard/src/lib/transfer.ts:551]: Added a `fail-usdc-transfer` entry to the recovery guide (pre-burn USDC stuck). 35f404ca

File: crates/dto/src/transfer.rs (hunk 4 above)
Comment (JuaniRios) [reply crates/dto/src/transfer.rs:388]: `src/lib/api/` is gitignored (generated ts-rs bindings, regenerated by the dashboard build/CI), so there's nothing to commit. Nothing imports `UsdcBridgeStatus` either — the panel uses a local shape — so there's no svelte-check break.

File: dashboard/src/lib/components/recovery-guide.svelte (hunk 10 above)
Comment (JuaniRios) [reply dashboard/src/lib/components/recovery-guide.svelte]: Added a shared `recoveryModeColor` helper with a third (sky) colour for `live-rpc-only`, wired into both the command block (badge) and the guide (text). 35f404ca

File: dashboard/src/lib/components/recovery-guide.svelte (hunk 11 above)
Comment (JuaniRios) [reply dashboard/src/lib/components/recovery-guide.svelte:54]: Added a prominent simulation-build banner (keyed on `getSimulateSourceId()`) telling the operator the `stox` commands aren't runnable there and to use the per-object modal commands. 35f404ca

File: dashboard/src/lib/components/transfer-panel.svelte (hunk 12 above)
Comment (JuaniRios) [reply dashboard/src/lib/components/transfer-panel.svelte:61]: `TransferEntry` already models all kind-specific data as flat optionals (`symbol`/`quantity`/`amount`/`direction`); `status.postBurn?` follows that pattern and `transferRecoveryCommands` dispatches on `kind` before reading it. Narrowing only `status` into a union would be inconsistent with the rest of the type for a non-bug.

File: dashboard/src/lib/transfer.ts (hunk 7 above)
Comment (JuaniRios) [reply dashboard/src/lib/transfer.ts]: Agreed -- `process_found_trade` only guards on `last_acknowledged_trade_id` (single slot) and can double-apply an older accounted fill. It's a pre-existing CLI-path weakness, not introduced here; T7 removes the one-click dashboard vector, and I've filed RAI-1157 to fail it closed on an already-recorded `(tx_hash, log_index)`.

**Review by 0xgleb — APPROVED** [review]
